src/NUnitTestExtractor/NUnitTestExtractorTests/TestExtractorTests.cs

```diff
@@ -0,0 +1,255 @@
+        /// <param name="level">The level to pass to GetTests()</param>
+        private static void InvokeGetTestsUsingFileOutput(StringBuilder builder, List<string> dlls, NUnitTestExtractorApp.Level level)
+        {
+            using (StringWriter writer = new StringWriter(builder, CultureInfo.CurrentCulture))
+            {
+                NUnitTestExtractorApp.GetTests(dlls, level, writer);
+            }
+        }
+
+        /// <summary>
+        /// Sets up and runs process which is used to read lines in the standard output
+        /// </summary>
+        /// <param name="dll">Dlls to pass to the process</param>
+        /// <param name="level"> level to pass to the process</param>
+        /// <returns></returns>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Reliability", "CA2000:Dispose objects before losing scope")]
+        private static Process SetUpProcess(string dll, string level)
+        {
+            Process process = new Process();
+
+            process.StartInfo.FileName = "NUnitTestExtractor.exe";
+            process.StartInfo.Arguments = "-l " + level + " " + dll;
+            process.StartInfo.UseShellExecute = false;
+            process.StartInfo.RedirectStandardOutput = true;
+
+            process.Start();
+
+            return process;
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1822:MarkMembersAsStatic")]
+        [TestCaseSource(typeof(InvalidDataTestCasesCollection))]
+        [Test]
+        public void WritingToFile_InvalidDll_NothingWritten(string dll, string level)
+        {
+            List<string> dlls = new List<string>();
+            dlls.Add(dll);
+            
+            StringBuilder builder = new StringBuilder();
+
+            InvokeGetTestsUsingFileOutput(builder, dlls, NUnitTestExtractorApp.ParseLevel(level));
+
+            Assert.That(builder.ToString(), Is.Empty);
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1822:MarkMembersAsStatic")]
+        [TestCaseSource(typeof(ValidDataTestCasesCollection))]
+        [Test]
+        public void WritingToFile_ValidDllAndEveryLevel_GetsWrittenWithProperFormat(string dll, string level)
+        {
+            List<string> dlls = new List<string>();
+            dlls.Add(dll);
+
+            StringBuilder builder = new StringBuilder();
+
+            InvokeGetTestsUsingFileOutput(builder, dlls, NUnitTestExtractorApp.ParseLevel(level));
+
+            string builderText = builder.ToString();
+            string firstLine = builderText.Substring(0,builderText.IndexOf(Environment.NewLine, StringComparison.OrdinalIgnoreCase) + 1);
+
+            Regex regex = null;
+
+            NUnitTestExtractorApp.Level myLevel = NUnitTestExtractorApp.ParseLevel(level);
+
+            switch (myLevel)
+            {
+                case NUnitTestExtractorApp.Level.Namespace:
+                    regex = new Regex(@"\w+\s\|\s[^\.]");
```

Comment:
Instead of looking for [^\.] which could be anything other than a . you could use \w+
Also, I usually start my regex with ^ and end with $ to verify that there's nothing unexpected before or after the regex match on the line.
Same with the others below, you could do \w+\.\w+ and \w+\.\w+\.\w